src/Smalot/PdfParser/Document.php

```diff
@@ -198,25 +221,36 @@ public function getFonts()
         return $this->getObjectsByType('Font');
     }
 
+    /**
+     * @return PDFObject
+     */
+    public function getFirstFont()
+    {
+        $fonts = $this->getFonts();
+
+        return reset($fonts);
+    }
+
     /**
      * @return Page[]
      *
      * @throws \Exception
      */
     public function getPages()
     {
-        if (isset($this->dictionary['Catalog'])) {
+        if ($this->hasObjectsByType('Catalog')) {
             // Search for catalog to list pages.
-            $id = reset($this->dictionary['Catalog']);
+            $catalogues = $this->getObjectsByType('Catalog');
+            $catalogue = reset($catalogues);
 
             /** @var Pages $object */
-            $object = $this->objects[$id]->get('Pages');
+            $object = $catalogue->get('Pages');
```

Comment:
`getObjectsByType` can return an empty array, therefore `$catalogue` is null and `$catalogue->get('Pages');` leads to an error, doesn't it? If so, please add a check up front.

Reply:
That is what the
```if ($this->hasObjectsByType('Catalog')) {```
checks beforehand, yes.

If we have objects by type `'Catalog'`, then we proceed to get these objects and do not worry if there aren't any (because we know there are). So there should always be at least one `$catalogue` there.